### DISCUSSION TITLE
Ephemeris to Calculate when the Sun crosses between Constellations?

### DISCUSSION THREAD

**Discussion by Aubstract** [Stellarium/stellarium#1972] [Ideas]
There have been a number of times I've wanted data on when the Sun crosses between constellations, and it doesn't seem possible in the Astronomical Calculations window. I've had to manually change the date while tracking the Sun to see when it crosses a border between constellations and write down the dates, but that's super time-consuming. It seems like it would be a fairly easy thing to implement, so I'm hoping that could happen. 

I really would love a data set of multiple years, so I could plot it against other data sets like birthrate (to help argue against astrology lol).

**Comment by gzotti** [comment]
You could implement this as script. Increase date by a full day, and if IAU constellation has changed, go back and do a binary search on that day until you have found the minute (or second). Then advance by days again to find the next. Results can be printed to output.

If you are talking about astrology, you would instead search for ecliptical longitude and when the sun reaches the next *sign* at a multiple of 30, else the same algorithm can be applied.
